event.go

```diff
@@ -55,3 +51,45 @@ func createTriggerPayload(channels []string, event string, data interface{}, soc
 		SocketId: socketID,
 	})
 }
+
+func createTriggerBatchPayload(batch []Event, encryptionKey string) ([]byte, error) {
+
+	for idx, e := range batch {
+
+		dataBytes, err := byteEncodePayload(e.Data)
+		if err != nil {
+			return nil, err
+		}
+
+		if isEncryptedChannel(e.Channel) {
+			batch[idx].Data = encrypt(e.Channel, dataBytes, encryptionKey)
+		} else {
+			batch[idx].Data = string(dataBytes)
+		}
+
+		if len(batch[idx].Data) > maxEventPayloadSize {
+			return nil, fmt.Errorf("Data of the event #%d in batch, must be smaller than 10kb", idx)
+		}
+
+	}
+
+	return json.Marshal(&batchRequest{batch})
+}
+
+func byteEncodePayload(data interface{}) ([]byte, error) {
```

Comment:
This can be renamed to just `encodePayload`. The output type is specified in the type signature already.

Reply:
Yes, good point